### ISSUE TITLE
MathJax custom options don't render correctly when run through GitHub, but work fine on localhost

### ISSUE THREAD

**Issue by ZiadHatab** [cotes2020/jekyll-theme-chirpy#900]
Hi!

I have spent hours now trying to debug this. I have updated the MathJax configuration in the `js-selector.html` file. The modification I did is similar to what the official website of MathJax mention https://docs.mathjax.org/en/latest/options/input/tex.html#tex-options 

My problem is that, the configuration I included works fine when I run the website as localhost, but when I run it through github pages, it  doesn't render correctly. 

Here is the modified MathJax section I use in `js-selector.html`:
```html
{% if page.math %}
  <!-- MathJax -->
  <script>
  /* see: <https://docs.mathjax.org/en/latest/options/input/tex.html#tex-options> */
  MathJax = {
    tex: {
      inlineMath: [              /* start/end delimiter pairs for in-line math */
        ['$','$'],
        ['\\(','\\)']
      ],
      displayMath: [             /* start/end delimiter pairs for display math */
        ['$$', '$$'],
        ['\\[', '\\]']
      ],
      processEscapes: true,      // use \$ to produce a literal dollar sign
      processEnvironments: true, // process \begin{xxx}...\end{xxx} outside math mode
      processRefs: true,         // process \ref{...} outside of math mode
      digits: /^(?:[0-9]+(?:\{,\}[0-9]{3})*(?:\.[0-9]*)?|\.[0-9]+)/,  // pattern for recognizing numbers
      tags: 'ams',               // 'none' or 'ams' or 'all'
      tagSide: 'right',          // side for \tag macros
      tagIndent: '0.8em',        // amount to indent tags
      useLabelIds: true,         // use label name rather than tag for ids
      maxMacros: 1000,           // maximum number of macro substitutions per expression
      maxBuffer: 5 * 1024,       // maximum size for the internal TeX string (5K)
      baseURL:                   // URL for use with links to tags (when there is a <base> tag in effect)
        (document.getElementsByTagName('base').length === 0) ?
          '' : (String(document.location).replace(/#.*$/, '')),
      formatError:               // function called when TeX syntax errors occur
          (jax, err) => jax.formatError(err),
      macros: {                // my maros that I usually use in texstudio
        Re:  ['\\operatorname{Re}\\left({#1}\\right)', 1], // real part with expandable parentheses
        Im:  ['\\operatorname{Im}\\left({#1}\\right)', 1], // imag part with expandable parentheses
        EV:  ['\\mathbb{E}\\left({#1}\\right)', 1],        // expected value
        bs:  ['\\boldsymbol{#1}', 1],                      // bold symbol. Also works with greek letters
        vc:  ['\\mathrm{vec}\\left({#1}\\right)', 1],      // vec() operator
        cov: ['\\mathrm{cov}\\left({#1}\\right)', 1],      // covariance operator
        var: ['\\mathrm{var}\\left({#1}\\right)', 1],      // variance operator
        std: ['\\mathrm{std}\\left({#1}\\right)', 1],      // standard deviation operator
      }
    }
  };
  </script>
  <script src="{{ site.data.assets[origin].polyfill.js | relative_url }}"></script>
  <script id="MathJax-script" async src="{{ site.data.assets[origin].mathjax.js | relative_url }}">
  </script>
{% endif %}
```

Here is the rendered website, when run on localhost:
![image](https://user-images.githubusercontent.com/46247580/220217217-b5babb25-f82f-436f-87ef-5551ab6f50ce.png)

Now, here is the website when run through github (https://ziadhatab.github.io/posts/trl-calibration/):
![image](https://user-images.githubusercontent.com/46247580/220217533-5024ee42-7400-4492-8923-b054c7f111d2.png)

Help would be appreciated~

**Comment by ZiadHatab** [comment]
So I decided to start from scratch and test if the problem was due to my modification or inherent in the project. I forked the original project and made a simple change in the MathJax options to include tags option (based on the example from MathJax: https://mathjax.github.io/MathJax-demos-web/equation-refs.html.html):
```java
MathJax = {
  tex: {
    tags: 'all',  // should be 'ams', 'none', or 'all'
    inlineMath: [              /* start/end delimiter pairs for in-line math */
      ['$','$'],
      ['\\(','\\)']
    ],
    displayMath: [             /* start/end delimiter pairs for display math */
      ['$$', '$$'],
      ['\\[', '\\]']
    ]
  }
};
```

And then I added a `\label{ref1}` and `\ref{ref1}` to the equation in the `2019-08-08-text-and-typography.md` post. 
```markdown
$$ \sum_{n=1}^\infty 1/n^2 = \frac{\pi^2}{6} \label{ref1}$$

Try to reference equation \ref{ref1}.
```
Unfortunately, I'm still having the same problem. On localhost via `bundle exec jekyll serve` everything works fine, but after deploying as github pages, the equation numbering and referencing doesn't work.

On localhost:
![image](https://user-images.githubusercontent.com/46247580/221370814-28c69906-4a49-4f8f-95af-5420e1afab3f.png)

On github:
![image](https://user-images.githubusercontent.com/46247580/221370792-fe71dec1-9cf4-4031-adc1-701622b1213b.png)

Could anyone test if you can reference any equation when running your chirpy website on github?

**Comment by cotes2020** [comment]
So, you've been struggling with this issue for 4 days, yet the answer has long been written in the project documentation. You've paid for your time by ignoring the documentation, so it's time to just give you a clue.

See: <https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/.github/CONTRIBUTING.md#modifying-javascript>

**Comment by ZiadHatab** [comment]
Man... thanks!

To be fair to myself, I didn't spend 4 days on it. When it wasn't working, I just posted about it. Hoping someone will help out. Now, over the weekend, I thought to debug it myself.

Also, I just wanted to use the template. Why would I even think to look under the contribution section?

Anyway, thanks again!

For those wondering what was the answer... basically don't use `//` to comment in the JavaScript sections. Better, just remove the comments.
